ext/random/tests/03_randomizer/user_unsafe.phpt

```diff
@@ -0,0 +1,81 @@
+--TEST--
+Random: Randomizer: User: Engine unsafe
+--FILE--
+<?php
+
+// Empty generator
+$randomizer = (new \Random\Randomizer(
+    new class () implements \Random\Engine {
+        public function generate(): string
+        {
+            return '';
+        }
+    }
+));
+
+try {
+    $randomizer->getInt(\PHP_INT_MIN, \PHP_INT_MAX);
+} catch (\RuntimeException $e) {
+    echo "{$e->getMessage()}\n";
+}
+
+try {
+    $randomizer->getBytes(1);
+} catch (\RuntimeException $e) {
+    echo "{$e->getMessage()}\n";
+}
+
+try {
+    $randomizer->shuffleArray(\range(1, 10));
+} catch (\RuntimeException $e) {
+    echo "{$e->getMessage()}\n";
+}
+
+try {
+    $randomizer->shuffleString('foobar');
+} catch (\RuntimeException $e) {
+    echo "{$e->getMessage()}\n";
+}
+
+// Infinite loop
+$randomizer = (new \Random\Randomizer(
+    new class () implements \Random\Engine {
+        public function generate(): string
+        {
+            return "\xff\xff\xff\xff\xff\xff\xff\xff";
+        }
+    }
+));
+
+try {
+    $randomizer->getInt(\PHP_INT_MIN, \PHP_INT_MAX);
+} catch (\RuntimeException $e) {
+    echo "{$e->getMessage()}\n";
+}
+
+try {
+    $randomizer->getBytes(1);
+} catch (\RuntimeException $e) {
+    echo "{$e->getMessage()}\n";
+}
```

Comment:
These are misleading, because they don't actually throw an Exception. I suggest that you `var_dump()` each value (even in the cases where they are expected to fail) to uniquely determine which ones fail, because then a line will be printed for each case:

```suggestion
try {
    var_dump($randomizer->getInt(\PHP_INT_MIN, \PHP_INT_MAX));
} catch (\RuntimeException $e) {
    echo "{$e->getMessage()}\n";
}

try {
    var_dump(bin2hex($randomizer->getBytes(1)));
} catch (\RuntimeException $e) {
    echo "{$e->getMessage()}\n";
}
```